File: cses/1643.py
#!/usr/bin/env python3
# https://cses.fi/problemset/task/1643/
i = input
i()
s = m =-2e9
for x in map(int, i().split()):
    s = max(x, s + x)
    m = max(s, m)
print(m)

File: cses/1643.cc
// https://cses.fi/problemset/task/1643/
#include <bits/stdc++.h>

using namespace std;
using ll = long long;

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);
  ll n, s = 0, m = -2000000000;
  cin >> n;
  vector<ll> a(n);
  for (int i = 0; i < n; i++) cin >> a[i];
  for (int i = 0; i < n; i++) {
    s = max(s + a[i], a[i]);
    m = max(m, s);
  }
  cout << m << "\n";
}
